extract_endpoint/src/extract_endpoint/endpoint.py

```diff
@@ -59,25 +59,26 @@ def upload_file() -> typing.Tuple[str, int]:
         flask.abort(HTTPStatus.BAD_REQUEST)
     if 'salt' not in flask.request.form:
         flask.abort(HTTPStatus.BAD_REQUEST)
-    if flask.request.form.get('filename', '') == '':
-        flask.abort(HTTPStatus.BAD_REQUEST)
     if 'file' not in flask.request.files:
         flask.abort(HTTPStatus.BAD_REQUEST)
     if 'timestamp' not in flask.request.form:
         flask.abort(HTTPStatus.BAD_REQUEST)
 
     file = flask.request.files['file']
-    file_contents = file.read()
-    file_as_string = base64.b64encode(file_contents).decode('utf-8')
     signature = crypt_utils.sign_string(salt=flask.request.form['salt'], key=App.config['SECRET_KEY'],
-                                        data=file_as_string)
+                                        data=base64.b64encode(file.read()).decode('utf-8'))
     if flask.request.form['signature'] != signature:
         flask.abort(HTTPStatus.BAD_REQUEST)
 
-    filename = utils.secure_filename(flask.request.form['filename'])
-
-    if not azure_utils.upload_bytes_to_azure(App.config['AZURE_COORDINATES'], file_contents, filename):
-        flask.abort(HTTPStatus.BAD_GATEWAY)
+    try:
+        file_z = zipfile.ZipFile(file)
```

Comment:
Presumably my fears will be laid to rest once I read through the tests, but hasn't the file position been advanced to the end of the file by the read on `69`?

Reply:
yeah, I wondered about that too, but it seems to work anyway. However, it wouldn't hurt to throw a seek(0) in there just in case this behaviour changes (or is different on other systems).